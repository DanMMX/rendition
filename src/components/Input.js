import styled, { withTheme } from 'styled-components'
import asRendition from '../asRendition'
import { monospace, px } from '../utils'
import { radius } from '../theme'
import { compose } from 'recompose'

const getColor = props => {
  if (props.invalid) {
    return props.theme.colors.danger.main
  }
  if (props.valid) {
    return props.theme.colors.success.main
  }
  if (props.disabled) {
    return props.theme.colors.gray.light
  }
  return props.theme.colors.gray.main
}

const Base = styled.input`
  border-radius: ${px(radius)};
  height: ${props =>
    px(props.emphasized ? props.theme.space[5] : props.theme.space[4])};
  font-size: inherit;
  border: 1px solid ${props => getColor(props)};
  padding: 0px 16px;

  &[disabled] {
    background-color: ${props => props.theme.colors.gray.light};
  }
  &:focus {
    border-color: ${props => props.theme.colors.secondary.main};
  }
  &:hover {
    box-shadow: 0 0 4px 1px rgba(0, 0, 0, 0.1);
  }
  &::placeholder {
    color: ${props => props.theme.colors.gray.main};
  }

  &[type="checkbox"] {
    height: auto;
    font-size: ${props => px(props.theme.fontSizes[1])};
  }

  ${monospace};
`

export default compose(withTheme, asRendition)(Base)
