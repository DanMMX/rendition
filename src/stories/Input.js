import { h } from 'preact'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'
import Input from '../components/Input'

const Container = styled.div`margin: 30px;`

storiesOf('Input', module)
  .addWithInfo('Standard', () => {
    return (
      <Container>
        <Input placeholder='Placeholder Text' />
      </Container>
    )
  })
  .addWithInfo('Emphasized', () => {
    return (
      <Container>
        <Input emphasized placeholder='Placeholder Text' />
      </Container>
    )
  })
