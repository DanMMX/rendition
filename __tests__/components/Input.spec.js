/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Input from '../../src/components/Input'

test('Input renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Input placeholder='Placeholder Text' />
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
