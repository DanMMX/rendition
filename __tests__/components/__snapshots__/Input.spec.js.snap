// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <input
    className="sc-htpNat fGEkbD sc-bwzfXH bicRBw"
    placeholder="Placeholder Text"
  />
</div>
`;
